tests/backup/backup_helper.go

```diff
@@ -1218,85 +1089,104 @@ func DeletePodWithLabelInNamespace(namespace string, label map[string]string) er
 	if err != nil {
-		return false, err
+		return err
 	}
-	return true, nil
+	return nil
 }
 
 // restoreSuccessCheck inspects restore task
-func restoreSuccessCheck(restoreName string, orgID string, retryDuration int, retryInterval int, ctx context.Context) (bool, error) {
+func restoreSuccessCheck(restoreName string, orgID string, retryDuration time.Duration, retryInterval time.Duration, ctx context.Context) error {
 	if retryDuration == 0 {
-		retryDuration = maxWaitPeriodForRestoreCompletionInMinute
+		retryDuration = maxWaitPeriodForRestoreCompletionInMinute * time.Minute
 	}
 	if retryInterval == 0 {
-		retryInterval = 30
+		retryInterval = 30 * time.Second
 	}
-	backupDriver := Inst().Backup
 	restoreInspectRequest := &api.RestoreInspectRequest{
 		Name:  restoreName,
 		OrgId: orgID,
 	}
-	restoreSuccessCheck := func() (interface{}, bool, error) {
+	statusesExpected := [...]api.RestoreInfo_StatusInfo_Status{
+		api.RestoreInfo_StatusInfo_PartialSuccess,
+		api.RestoreInfo_StatusInfo_Success,
+	}
+	statusesUnexpected := [...]api.RestoreInfo_StatusInfo_Status{
+		api.RestoreInfo_StatusInfo_Invalid,
+		api.RestoreInfo_StatusInfo_Aborted,
+		api.RestoreInfo_StatusInfo_Failed,
+	}
+	_restoreSuccessCheck := func() (interface{}, bool, error) {
 		resp, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
 		if err != nil {
 			return "", false, err
 		}
-		restoreResponseStatus := resp.GetRestore().GetStatus()
-		if restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_PartialSuccess || restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_Success {
-			log.Infof("Restore status for [%s] - %s", restoreName, restoreResponseStatus.GetStatus())
-			return "", false, nil
+		actual := resp.GetRestore().GetStatus().Status
+		reason := resp.GetRestore().GetStatus().Reason
+		for _, status := range statusesExpected {
+			if actual == status {
+				return "", false, nil
+			}
 		}
-		return "", true, fmt.Errorf("expected status of %s - [%s] or [%s], but got [%s]",
-			restoreName, api.RestoreInfo_StatusInfo_PartialSuccess.String(), api.RestoreInfo_StatusInfo_Success, restoreResponseStatus.GetStatus())
-	}
-	_, err := task.DoRetryWithTimeout(restoreSuccessCheck, time.Duration(retryDuration)*time.Minute, time.Duration(retryInterval)*time.Second)
-	if err != nil {
-		return false, err
-	}
-	restoreInspectRequest = &api.RestoreInspectRequest{
-		Name:  restoreName,
-		OrgId: orgID,
+		for _, status := range statusesUnexpected {
+			if actual == status {
+				return "", false, fmt.Errorf("restore status for [%s] expected was [%v] but got [%s] because of [%s]", restoreName, statusesExpected, actual, reason)
+			}
+		}
+		return "", true, fmt.Errorf("restore status for [%s] expected was [%v] but got [%s] because of [%s]", restoreName, statusesExpected, actual, reason)
 	}
-
-	resp, err := backupDriver.InspectRestore(ctx, restoreInspectRequest)
+	time.Sleep(retryInterval)
```

Comment:
why we need a sleep time here?

Reply:
Since success checks are typically called after creation and immediate inspection is not very useful, one approach could be to wait for a retry interval using sleep and then try again.

Reply:
Not required. We can start checking for success immediately. You can remove this from both `backupSuccessCheck` and `restoreSuccessCheck`

Reply:
Done.